### ISSUE TITLE
getdir does not work as specified

### ISSUE THREAD

**Issue by mauro3** [helgee/RemoteFiles.jl#13]
`getdir` does not return the right path if `@RemoteFile` is used in a sub-sub directory of the project, e.g. when called in `SomeProject/src/subdir/j.jl` it will return `SomeProject/src/data` instead of `SomeProject/data`.  Probably using some function of `Pkg` could resolve that.

**Comment by mauro3** [comment]
Maybe `Pkg.project().path`?

**Comment by helgee** [comment]
Thanks for bringing this up. Unfortunately, I never had a good solution for this. A workaround would be to use the explicit `dir=` keyword argument when constructing a `RemoteFile`.

In a post-1.0 world (and especially post-1.3), `getdir` would need to handle two scenarios:

- The `RemoteFile` belongs to a project: Then `Pkg.project().path` would be correct.
- The `RemoteFile` belongs to a package: Then the path should probably be something like `.julia/data/$PackageName/$hash/$file`. AFAIK there is no public function to get such a path but I might be wrong.

Is there a way to distinguish between the two?

I actually hope that the Artifacts system will make this package obsolete in the near future (https://github.com/JuliaLang/Pkg.jl/issues/796#issuecomment-523154714). Maybe @staticfloat can comment on that?

**Comment by mauro3** [comment]
I would be fine with an explicit `dir`.  Also, this might be used in just non-project/package scripts (ok, that shouldn't be done but still).

**Comment by helgee** [comment]
> I would be fine with an explicit `dir`.

That should already work: https://helgeeichhorn.de/RemoteFiles.jl/stable/api/#RemoteFiles.@RemoteFile-Tuple{Symbol,Any,Vararg{Any,N}%20where%20N}

**Comment by mauro3** [comment]
Yep, but making it mandatory (as opposed non-mandatory but buggy) would probably be better.

**Comment by helgee** [comment]
Ah, now I get it. Yeah, that makes sense 😂

**Comment by mauro3** [comment]
That would also pave the way for my other suggestion: #14
